### DISCUSSION TITLE
Bar Chart: Labels

### DISCUSSION THREAD

**Discussion by errantmind** [f5/unovis#204] [Q&A] [answered]
Hello, I have a horizontal bar chart and wish to add labels on each individual bar (towards the end) with some extra data. You can see what I mean [here](https://observablehq.com/@observablehq/plot-horizontal-bar-chart-with-label?intent=fork). 

Is this possible in Unovis? Sorry if I missed it somewhere.

**Comment by rokotyan** [comment, marked as answer]
@errantmind Yes, you can achieve that by using the `XYLabels` component. We don't have docs for it yet, but here's a brief code snippet for React to give you an example:
```jsx
<VisXYContainer>
  <VisStackedBar data={data} x={d => d.x} y={accessors} />
  <VisXYLabels
    data={data}
    y={d => d.y / 2}
    x={d => d.x}
    label={d => `${d.y.toFixed(1)} bps`}
  />
  <VisAxis type='x' numTicks={10}/>
  <VisAxis type='y' tickFormat={(y: number) => `${y}bps`}/>
</VisXYContainer>
```

<img width="1048" alt="image" src="https://github.com/f5/unovis/assets/755708/fb13be5a-81ba-44a8-bde5-aced602e72e4">

**Comment by errantmind** [comment]
Thanks! I'll give it a shot.

**Comment by conradfuhrman** [comment]
Hi all, we're just jumping into this library and are getting some very confusing results with the XYLabel component. We are using Vue3 not TS, but in a script setup. 

We are using this for our Labels:
```
<VisXYLabels :data="formatLabelData(visitorsPerHour)" :y="(d) => d.y / 2" :x="(d) => d.x" :label="d => d.y" clustering="false" />
```

We've verified that the data is formatting properly and returning what we expect.

```
const formatLabelData = (visitorsPerHour) => {
  let labels = Object.entries(visitorsPerHour).map(([key, value]) => ({ x: Number(key), y: Number(value) })).filter(entry => entry.y !== 0)
  return labels
}
```

We've also tried using our data that we supply to the container but always get the same results. It appears it's clustering, but for a strange reason/position and I'm not sure why as you can see below. I'm not sure if there is an error with the Vue side of things, but we just can't seem to get this to work.

![Screenshot 2024-02-19 at 2 50 28 PM](https://github.com/f5/unovis/assets/455824/bc5638a4-b358-4549-9e64-b1c62a810253)

**Reply by rokotyan** [reply]
@conradfuhrman Can you try disabling clustering and write what happened? Also, a CodeSandbox example that reproduces the problem might be helpful.

**Comment by rokotyan** [comment]
Thanks, glad to hear that!

Yes, a CodeSandbox would be very helpful as it would save a ton of time in reproducing the problem.

**Reply by crllnsmnn** [reply]
You can find a sandbox [here](https://stackblitz.com/edit/vue-qzqeyr?file=src%2FApp.vue%3AL10).
When I use a static label (e.g. label="Test"), I can see three labels, but when I make it dynamic (:label="(d) => d.x") it seems to be clustering labels. Maybe there is something else I am doing wrong?

**Reply by rokotyan** [reply]
Thanks @crllnsmnn! 

We've found an issue on our side and planning to fix it soon. I'll keep you posted
